src/fsharp/ConstraintSolver.fs

```diff
@@ -835,10 +835,26 @@ and SolveAnonInfoEqualsAnonInfo (csenv: ConstraintSolverEnv) m2 (anonInfo1: Anon
     (match anonInfo1.Assembly, anonInfo2.Assembly with 
         | ccu1, ccu2 -> if not (ccuEq ccu1 ccu2) then ErrorD (ConstraintSolverError(FSComp.SR.tcAnonRecdCcuMismatch(ccu1.AssemblyName, ccu2.AssemblyName), csenv.m,m2)) else ResultD ()
         ) ++ (fun () -> 
+
     if not (anonInfo1.SortedNames = anonInfo2.SortedNames) then 
-        let namesText1 = sprintf "%A" (Array.toList anonInfo1.SortedNames)
-        let namesText2 = sprintf "%A" (Array.toList anonInfo2.SortedNames)
-        ErrorD (ConstraintSolverError(FSComp.SR.tcAnonRecdFieldNameMismatch(namesText1, namesText2), csenv.m,m2)) 
+        let (|Subset|Superset|Overlap|CompletelyDifferent|) (first, second) =
+            let first = Set first
+            let second = Set second
+            let secondOnly = (second - first) |> Set.toList
+            let firstOnly = (first - second) |> Set.toList
```

Comment:
Another style nit but it would be good to put a newline after this

Reply:
Done.